tests/bones/test_uri_bone.py

```diff
@@ -0,0 +1,127 @@
+        url_value = "http://localhost:20"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "http://localhost:21"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.is_invalid(res, url_value)
+
+        # Test domain_allowed_list
+        bone = UriBone(domain_allowed_list=["viur.dev"])
+        url_value = "https://www.viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "https://foo.viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "https://viur.com/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.is_invalid(res, url_value)
+
+        # Test for fnmatch in domain_allowed_list
+        bone = UriBone(domain_allowed_list=["w*.viur.dev"])
+        url_value = "https://www.viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "https://www2.viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "https://foo.viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.is_invalid(res, url_value)
+        url_value = "https://viur.dev/"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.is_invalid(res, url_value)
+
+        # Test for clean_get_params
+        bone = UriBone(clean_get_params=False)
+        url_value = "https://www.viur.dev/?foo=bar"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+
+        bone = UriBone(clean_get_params=True)
+        url_value = "https://www.viur.dev/?foo=bar"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual(("https://www.viur.dev/", None), res)
+
+        # Test for local_path_allowed
+        bone = UriBone(local_path_allowed=True)
+        url_value = "/foo/bar/?a=b"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual((url_value, None), res)
+        url_value = "foo/bar/?a=b"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.assertEqual(("/foo/bar/?a=b", None), res)
+
+        bone = UriBone(local_path_allowed=False)
+        url_value = "/foo/bar/?a=b"
+        res = bone.singleValueFromClient(url_value, skel, self.bone_name, None)
+        self.is_invalid(res, url_value)
+
+        # Test for general schema, not valid URLs
```

Comment:
I added some more tests for the schema, they seems to be identical to JS: https://jsfiddle.net/y03e26p5/5/